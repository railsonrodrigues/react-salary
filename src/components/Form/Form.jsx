import React from 'react';
import './../../globalStyles.css';

import Input from "../Input/Input";
import Button from "../Button/Button";

export default function Form({ onSubmit, className }) {
  return (
    <form onSubmit={onSubmit} className={className}>
      <Input id="grossSalary" placeholder="Salário bruto" />
      <Input id="nDependents" placeholder="Número de dependentes" />
      <Input id="otherDiscounts" placeholder="Outros Descontos" />
      <Button value="Calcular" />
    </form>
  );
}
