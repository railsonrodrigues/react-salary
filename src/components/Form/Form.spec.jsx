import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
import Form from './Form';

describe('<Form />', () => {
  it('should render Form', () => {
    const fn = jest.fn();

    render(<Form onSubmit={fn} role="form"/>);
    const form = screen.getByRole('form');
    expect(form).toBeInTheDocument();
  })

  it('should render input fields', () => {
    const fn = jest.fn();

    expect.assertions(3);

    render(<Form onSubmit={fn} role="form"/>);
    const input = screen.getAllByRole('textbox');
    input.forEach(input => {
      expect(input).toBeInTheDocument();
    });
  })

  it('should render button', () => {
    const fn = jest.fn();

    expect.assertions(1);

    render(<Form onSubmit={fn} role="form"/>);
    const button = screen.getByRole('button', {name: /calcular/i});
    expect(button).toBeInTheDocument();
  })
  
  it('should call function on submit', () => {
    const fn = jest.fn();

    render(<Form onSubmit={fn} role="form"/>);
    const form = screen.getByRole('form');

    fireEvent.submit(form);
    expect(fn).toHaveBeenCalledTimes(1);
  })
})