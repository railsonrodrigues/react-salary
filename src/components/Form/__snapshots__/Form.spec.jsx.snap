// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match snapshot 1`] = `
<form
  role="form"
>
  <div
    class="sc-dlnjwi hgOEXf"
  >
    <input
      class="sc-bdnxRM dBDSSv"
      id="grossSalary"
      placeholder="Salário bruto"
      type="text"
    />
    <input
      class="sc-bdnxRM dBDSSv"
      id="nDependents"
      placeholder="Número de dependentes"
      type="text"
    />
    <input
      class="sc-bdnxRM dBDSSv"
      id="otherDiscounts"
      placeholder="Outros Descontos"
      type="text"
    />
  </div>
  <div
    class="sc-hKFxyN fGSqKZ"
  >
    <button
      class="sc-gtsrHT gpeqvW"
    >
      Calcular
    </button>
  </div>
</form>
`;
